dracut.sh

```diff
@@ -2453,6 +2464,24 @@ fi
 
 umask 077
 
+sbat_default="sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md"
+sbat_out=$DRACUT_TMPDIR/uki.sbat
+
+clean_sbat_string() {
+    inp=$1
+    temp=$DRACUT_TMPDIR/temp.sbat
+    sed "s|$sbat_default||g" "$inp" > "$temp"
+    awk -i inplace 'NF' "$temp"
```

Comment:
Is the `awk` required? Would not be enough `sed -i "/${sbat_default//\//\\/}/d" "$temp"` ?

Reply:
I am not bash expert, but your line doesn't seem to do what I wanted. I need to get rid of the unnecessary newlines that are left when removing SBAT_DEFAULT, and I found online that `awk` does the job :) Feel free to suggest any alternative command line, as long as it works I have no issue in changing the command.

Reply:
The `sed` I suggested removes lines matching the pattern (no new empty lines are left).

```
localhost:~ # objcopy -O binary --only-section=.sbat /usr/lib/systemd/boot/efi/linuxx64.efi.stub stub.sbat
localhost:~ # cat stub.sbat 
sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md
systemd,1,The systemd Developers,systemd,253,https://systemd.io/
systemd.opensuse,1,The openSUSE Project,systemd,253.1-1.2,mailto:security@suse.de
localhost:~ # sbat_default="sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md"
localhost:~ # sed -i "/${sbat_default//\//\\/}/d" stub.sbat
localhost:~ # cat stub.sbat
systemd,1,The systemd Developers,systemd,253,https://systemd.io/
systemd.opensuse,1,The openSUSE Project,systemd,253.1-1.2,mailto:security@suse.de
```

Reply:
you are absolutely right, sorry for the confusion. I'll use your command instead of awk